src/backend/common/helpers/insights_leaderboard_team_helper.py

```diff
@@ -1,262 +1,308 @@
-                continue
+        winners = set()
+        for award in event.awards:
+            if award.award_type_enum == AwardType.WINNER:
+                for team_key in award.team_list:
+                    self.active_streaks[team_key.string_id()] += 1
+                    self.longest_streaks[team_key.string_id()] = max(
+                        self.longest_streaks[team_key.string_id()],
+                        self.active_streaks[team_key.string_id()],
+                    )
+                    winners.add(team_key.string_id())
+
+        for team in event.teams:
+            if team.key_name not in winners and self.active_streaks[team.key_name] > 0:
+                self.active_streaks[team.key_name] = 0
+
+    def make_insight(self, year: Year) -> Optional[Insight]:
+        if year != 0:
+            return None
 
-            winners = set()
-            for award in event.awards:
-                if award.award_type_enum == AwardType.WINNER:
-                    for team_key in award.team_list:
-                        active_streaks[team_key.string_id()] += 1
-                        longest_streaks[team_key.string_id()] = max(
-                            longest_streaks[team_key.string_id()],
-                            active_streaks[team_key.string_id()],
-                        )
-                        winners.add(team_key.string_id())
-
-            for team in event.teams:
-                if team.key_name not in winners and active_streaks[team.key_name] > 0:
-                    active_streaks[team.key_name] = 0
+        for team_key, active_streak in self.active_streaks.items():
+            if active_streak > 0:
+                self.longest_streaks[team_key] = max(
+                    self.longest_streaks[team_key], active_streak
+                )
 
         return make_leaderboard_from_dict_counts(
-            {k: v for k, v in longest_streaks.items() if v > 1},
+            {k: v for k, v in self.longest_streaks.items() if v > 1},
             Insight.TYPED_LEADERBOARD_LONGEST_QUALIFYING_EVENT_STREAK,
-            arguments.year,
+            year,
         )
+
+
+class InsightsLeaderboardTeamCalculator:
+    @staticmethod
+    def make_insights(
+        year: Year, calculators: Optional[List[AbstractLeaderboardCalculator]] = None
+    ) -> List[Insight]:
+        if calculators is None:
+            calculators = [
+                MostBlueBannersCalculator(),
+                MostAwardsCalculator(),
+                MostNonChampsEventWinsCalculator(),
+                MostMatchesPlayedCalculator(),
+                MostEventsPlayedAtCalculator(),
+                MostUniqueTeamsPlayedWithOrAgainstCalculator(),
+                MostNonChampsImpactWinsCalculator(),
+                MostWffasCalculator(),
+                LongestEinsteinStreakCalculator(),
+                LongestQualifyingEventStreakCalculator(),
+            ]
+
+        args = make_leaderboard_args(year)
```

Comment:
I don't think we need `make_leaderboard_args` anymore? or at least not the part that turns the events/awards into generators.